### PR TITLE
[backport] Write guides to implement new-style functions

### PR DESCRIPTION
**Pull request by toslunar**
Backport of #4986

### CODE REVIEW

**Comment by toslunar** [comment]
[automatic post] Jenkins, test this please.

File: docs/source/guides/functions.rst [hunk 1]
```diff
@@ -17,57 +17,112 @@ After reading this section, you will be able to:
...
 
-   class MulAdd(Function):
+   class MulAdd(FunctionNode):
        def forward_cpu(self, inputs):
+           # Unpack input arrays (``numpy.ndarray``).
            x, y, z = inputs
+
+           # Mark inputs (``x`` and ``y``) as retained so that it can be
+           # accessed during the backward process.
+           selt.retain_inputs((0, 1))
```
Comment (asi1024) [inline docs/source/guides/functions.rst]:
```suggestion
           self.retain_inputs((0, 1))
```

Comment (kmaehashi) [reply docs/source/guides/functions.rst]: Thanks, made another PR to fix this in master branch. #6734

**Comment by kmaehashi** [comment]
Let's merge this along with backport of #6734.

**Comment by kmaehashi** [comment]
Jenkins, test this please.

**Comment by kmaehashi** [comment]
LGTM!

**Pull request merged**